runtime/src/accounts_index.rs

```diff
@@ -581,6 +622,36 @@ impl<T: 'static + Clone> AccountsIndex<T> {
         max_root
     }
 
+    pub fn update_secondary_indexes(
+        &self,
+        slot: Slot,
+        pubkey: &Pubkey,
+        account_owner: &Pubkey,
+        account_data: &[u8],
+    ) {
+        if *account_owner == inline_spl_token_v2_0::id() && account_data.len() >= PUBKEY_BYTES {
```

Comment:
I think we also need to check that `account_data.len() == spl_token_v2_0::state::Account::LEN`, otherwise this will incorrectly index Mint and Multisig accounts as well